charts/consul/values.yaml

```diff
@@ -2458,6 +2458,111 @@ terminatingGateways:
   gateways:
     - name: terminating-gateway
 
+# Conffiguration setings for the Consul API Gateway integration
+apiGateway:
+  # When true the helm chart will install the Consul API Gateway controller
+  enabled: false
+
+  # Image to use for the api-gateway-controller pods and gateway instances
+  # @type: string
+  image: null
+
+
+  # Override global log verbosity level for api-gateway-controller pods. One of "debug", "info", "warn", or "error".
+  # @type: string
+  logLevel: info
+  managedGatewayClass:
+    enabled: true
+    # This value defines `nodeSelector` (https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector)
+    # labels for gateway pod assignment, formatted as a multi-line string.
+    #
+    # Example:
+    #
+    # ```yaml
+    # nodeSelector: |
+    #   beta.kubernetes.io/arch: amd64
+    # ```
+    #
+    # @type: string
+    nodeSelector: null
+
+    # This value defines the type of service created for gateways (e.g. LoadBalancer, ClusterIP)
+    serviceType: LoadBalancer
```

Comment:
If the service is of type NodePort, would you ever need to also accept a user-provided node port? If so, it might be better to convert it a `service` stanza and then have `type` under it. We do something like this for example for admin partitions:

https://github.com/hashicorp/consul-k8s/blob/ca980cbec061dd08bb0534deaa2b3352a891924c/charts/consul/values.yaml#L46-L73

Reply:
In this case no. It is driving the configuration of our GatewayClassConfig. The user provides the port in the Gateway definition. We simply have a toggle for if those ports should be mapped to node ports of the same value. 

Reference:
https://github.com/hashicorp/consul-api-gateway/blob/c82e707fb30bebf1a88980568e0f67e318f65496/pkg/apis/v1alpha1/types.go#L61-L67